src/topology/subset_properties.lean

```diff
@@ -89,61 +89,87 @@ lemma compact_iff_ultrafilter_le_nhds {s : set α} :
   in
   have f ≠ ⊥, from infi_ne_bot_of_directed ⟨a⟩
-    (assume ⟨c₁, hc₁, hc'₁⟩ ⟨c₂, hc₂, hc'₂⟩, ⟨⟨c₁ ∪ c₂, union_subset hc₁ hc₂, finite_union hc'₁ hc'₂⟩,
-      principal_mono.mpr $ diff_subset_diff_right $ sUnion_mono $ subset_union_left _ _,
-      principal_mono.mpr $ diff_subset_diff_right $ sUnion_mono $ subset_union_right _ _⟩)
-    (assume ⟨c', hc'₁, hc'₂⟩, show principal (s \ _) ≠ ⊥, by simp only [ne.def, principal_eq_bot_iff, diff_eq_empty]; exact h hc'₁ hc'₂),
-  have f ≤ principal s, from infi_le_of_le ⟨∅, empty_subset _, finite_empty⟩ $
-    show principal (s \ ⋃₀∅) ≤ principal s, from le_principal_iff.2 (diff_subset _ _),
+  (assume t₁ t₂, ⟨t₁ ∪ t₂,
+    principal_mono.mpr $ diff_subset_diff_right $
+      bUnion_subset_bUnion_left $ finset.subset_union_left _ _,
+    principal_mono.mpr $ diff_subset_diff_right $
+      bUnion_subset_bUnion_left $ finset.subset_union_right _ _⟩)
+   (assume t, show principal (s \ _) ≠ ⊥,
+     by simp only [ne.def, principal_eq_bot_iff, diff_eq_empty]; exact h _),
+  have f ≤ principal s, from infi_le_of_le ∅ $
+    show principal (s \ _) ≤ principal s, from le_principal_iff.2 (diff_subset _ _),
   let
     ⟨a, ha, (h : f ⊓ 𝓝 a ≠ ⊥)⟩ := hs f ‹f ≠ ⊥› this,
-    ⟨t, ht₁, (ht₂ : a ∈ t)⟩ := hc₂ ha
+    ⟨_, ⟨i, rfl⟩, (ha : a ∈ U i)⟩ := hsU ha
   in
-  have f ≤ principal (-t),
-    from infi_le_of_le ⟨{t}, by rwa singleton_subset_iff, finite_insert _ finite_empty⟩ $
-      principal_mono.mpr $
-        show s - ⋃₀{t} ⊆ - t, begin rw sUnion_singleton; exact assume x ⟨_, hnt⟩, hnt end,
-  have is_closed (- t), from is_open_compl_iff.mp $ by rw lattice.neg_neg; exact hc₁ t ht₁,
-  have a ∈ - t, from is_closed_iff_nhds.mp this _ $ ne_bot_of_le_ne_bot h $
-    le_inf inf_le_right (inf_le_left_of_le ‹f ≤ principal (- t)›),
-  this ‹a ∈ t›
-
+  have f ≤ principal (- U i),
+    from infi_le_of_le {i} $ principal_mono.mpr $ show s - _ ⊆ - U i, by simp [diff_subset_iff],
+  have is_closed (- U i), from is_open_compl_iff.mp $ by rw lattice.neg_neg; exact hUo i,
+  have a ∈ - U i, from is_closed_iff_nhds.mp this _ $ ne_bot_of_le_ne_bot h $
+    le_inf inf_le_right (inf_le_left_of_le ‹f ≤ principal (- U i)›),
+  this ‹a ∈ U i›
+
+/-- For every family of closed sets whose intersection avoids a compact set,
+there exists a finite subfamily whose intersection avoids this compact set. -/
+lemma compact.elim_finite_subfamily_closed {s : set α} {ι : Type u} (hs : compact s)
+  (Z : ι → set α) (hZo : ∀i, is_closed (Z i)) (hsZ : s ∩ (⋂ i, Z i) = ∅) :
+  ∃ t : finset ι, s ∩ (⋂ i ∈ t, Z i) = ∅ :=
+let ⟨t, ht⟩ := hs.elim_finite_subcover (λ i, - Z i) hZo
+  (by simpa only [subset_def, not_forall, eq_empty_iff_forall_not_mem, set.mem_Union,
+    exists_prop, set.mem_inter_eq, not_and, iff_self, set.mem_Inter, set.mem_compl_eq] using hsZ)
+    in
+⟨t, by simpa only [subset_def, not_forall, eq_empty_iff_forall_not_mem, set.mem_Union,
+    exists_prop, set.mem_inter_eq, not_and, iff_self, set.mem_Inter, set.mem_compl_eq] using ht⟩
```

Comment:
Could you please also prove https://en.wikipedia.org/wiki/Cantor's_intersection_theorem#Topological_Statement (may by for a `directed` family instead of a monotone one)? It seems like a trivial corollary now.

Reply:
Done.